parameter/BinaryStream.cpp

```diff
@@ -151,33 +149,40 @@ void CBinaryStream::reset()
     _uiPos = 0;
 }
 
+void CBinaryStream::assertValidAccess(size_t offset, size_t size) const
+{
+	assert(offset + size <= getSize());
+}
+
 void CBinaryStream::write(const uint8_t* puiData, size_t uiSize)
 {
-    assert(_uiPos + uiSize <= _uiDataSize);
+    assertValidAccess(_uiPos, uiSize);
```

Comment:
> std::copy will throw an exception in case of invalid access.
Are you sure ? I have never eared of such thing. I do not know any iterator that is required to throw on invalid access.